common/config.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"encoding/json"
+	"errors"
+
+	baseconfig "github.com/coreos/rkt/pkg/config"
+)
+
+const (
+	// Configuration directory basenames, refer to the docs of
+	// pkg/config to learn about their meaning. For current
+	// config, CDB is empty, because the configuration subdirs
+	// (auth.d, paths.d, etc.) are direct subdirectories of a
+	// toplevel config directory (/etc/rkt, /usr/lib/rkt, etc.).
+	CDB = ""
+)
+
+func NewConfigDirectory(cdb string) *baseconfig.Directory {
+	return baseconfig.NewDirectory(cdb, &rktJSON{})
+}
+
+type rktJSONConfigHeader struct {
+	RktVersion string `json:"rktVersion"`
+	RktKind    string `json:"rktKind"`
+}
+
+type rktJSON struct{}
+
+var _ baseconfig.Type = (*rktJSON)(nil)
+
+func (*rktJSON) Extension() string {
+	return "json"
+}
+
+func (*rktJSON) GetKindAndVersion(raw []byte) (string, string, error) {
+	var header rktJSONConfigHeader
```

Comment:
I think I'll keep it like it is now. It is not exported anyway. But later we might want to actually export it for `rkt config`, so it could be used for example like:

```
type authV1 struct {
	common.RktJSONConfigHeader
	…
}
```

I have some ideas, need to write them down.